src/Microsoft.Diagnostics.ExtensionCommands/DumpAsyncCommand.cs

```diff
@@ -0,0 +1,1159 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+using Microsoft.Diagnostics.DebugServices;
+using Microsoft.Diagnostics.Runtime;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.Diagnostics.ExtensionCommands
+{
+    [Command(Name = CommandName, Help = "Displays information about async \"stacks\" on the garbage-collected heap.")]
+    public sealed class DumpAsyncCommand : ExtensionCommandBase
+    {
+        /// <summary>Indent width.</summary>
+        private const int TabWidth = 2;
+        /// <summary>The name of the command.</summary>
+        private const string CommandName = "dumpasync";
+        /// <summary>The command invocation syntax when used in Debugger Markup Language (DML) commands.</summary>
+        private const string DmlCommandInvoke = $"!ext {CommandName}";
+
+        /// <summary>The help text to render when asked for help.</summary>
+        private static readonly string s_detailedHelpText =
+            "Usage: dumpasync  [--stats] [--coalesce] [--address <object address>] [--methodtable <mt address>] [--type <partial type name>] [--tasks] [--completed] [--fields]" + Environment.NewLine +
+            Environment.NewLine +
+            "Displays information about async \"stacks\" on the garbage-collected heap. Stacks" + Environment.NewLine +
+            "are synthesized by finding all task objects (including async state machine box" + Environment.NewLine +
+            "objects) on the GC heap and chaining them together based on continuations." + Environment.NewLine +
+            Environment.NewLine +
+            "Options:" + Environment.NewLine +
+            "  --stats        Summarize all async frames found rather than showing detailed stacks." + Environment.NewLine +
+            "  --coalesce     Coalesce stacks and portions of stacks that are the same." + Environment.NewLine +
+            "  --address      Only show stacks that include the object with the specified address." + Environment.NewLine +
+            "  --methodtable  Only show stacks that include objects with the specified method table." + Environment.NewLine +
+            "  --type         Only show stacks that include objects whose type includes the specified name in its name." + Environment.NewLine +
+            "  --tasks        Include stacks that contain only non-state machine task objects." + Environment.NewLine +
+            "  --completed    Include completed tasks in stacks." + Environment.NewLine +
+            "  --fields       Show fields for each async stack frame.";
```

Comment:
maybe add examples of methodtable and type? whenever I use SOS I find those helpful so I get the syntax right.

Reply:
Ok